test/feature-benchmark/scenarios.py

```diff
@@ -167,6 +168,36 @@ def benchmark(self) -> MeasurementSource:
         )
 
 
+class FastPathLimit(FastPath):
+    """Benchmark the case SELECT * FROM source LIMIT <i> , optimized by #21615"""
+
+    def init(self) -> list[Action]:
+        return [
+            TdAction(
+                f"""
+                $[version>7000] postgres-execute connection=postgres://mz_system:materialize@${{testdrive.materialize-internal-sql-addr}}
+                ALTER SYSTEM SET persist_fast_path_limit = 1000;
+
+                > CREATE MATERIALIZED VIEW v1 AS SELECT * FROM generate_series(1, {self.n()})
+                """
+            ),
+        ]
+
+    def benchmark(self) -> MeasurementSource:
+        return Td(
+            dedent(
+                """
+                > SELECT 1;
+                  /* A */
```

Comment:
What is this comment for? Is this a hint?

Reply:
https://github.com/MaterializeInc/materialize/blob/main/doc/developer/feature-benchmark.md#caveats-and-considerations
> Note that the framework will capture the time elapsed between the end of the query marked with /* A */ and the end of the query marked with /* B */. If you are benchmarking a single query, put the /* A */ marker on a dummy SELECT 1 query as shown in the example above.

Reply:
Yes, the way the feature benchmark operates is:
- testdrive runs each statement, and outputs the timestamp when it completed
- the feature benchmark then calculates the time elapsed between the end of the statement tagged with `/* A */` and the end of the statement tagged with `/* B */`

This approach was chosen so that:
- the SQL statements are run by testdrive, that is, Rust code, rather than by e.g. the pure-python Postgres driver, so hopefully no overhead from there. I had bad experience executing fast SQL from within Python in a previous job.
- one can benchmark an entire block of statements, not just a single one .
- the full testdrive syntax can be used to specify the workload
- there could be preparation and teardown statements outside of the `A..B` block and they will not be benchmarked.
- no need to write a part of the benchmark framework in Rust and minimal changes to testdrive -- testdrive only outputs a timestamp at statement completion.

But mostly because I was lazy at the time.

If we are benchmarking only a single statement, we do a parasitic `SELECT 1; /* A */`  at the start so that testdrive outputs a stating timestamp for the workload/statement.